src/Facade/Services/SACoreTypeService.cs

```diff
@@ -0,0 +1,42 @@
+﻿namespace Linn.Products.Facade.Services
+{
+    using System;
+
+    using Linn.Products.Domain.Linnapps;
+    using Linn.Products.Persistence.Linnapps;
+    using Linn.Products.Resources;
+
+    public class SaCoreTypeService : FacadeService<SaCoreType, int, SaCoreTypeResource>
+    {
+        private readonly IRepository<SaCoreType, int> repository;
+
+        public SaCoreTypeService(IRepository<SaCoreType, int> repository) : base(repository)
+        {
+            this.repository = repository;
+        }
+
+
+        protected override SaCoreType CreateFromResource(SaCoreTypeResource resource)
+        {
+            var coreType = new SaCoreType(
+                resource.CoreType,
+                resource.Description,
+                DateTime.Parse(resource.DateInvalid),
+                resource.LookAheadDays,
+                resource.SortOrder,
+                resource.TriggerLevel);
+
+            return coreType;
+        }
+
+        protected override void UpdateFromResource(SaCoreType saCoreType, SaCoreTypeResource resource)
+        {
+            saCoreType.Update(
+                resource.Description,
+                DateTime.Parse(resource.DateInvalid),
```

Comment:
Do you have a test for a null DateInvalid being passed into this just in case?
Maybe you do - I haven't checked